core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -521,6 +521,17 @@ private[spark] object SparkHadoopUtil extends Logging {
         SOURCE_SPARK_HADOOP)
     }
     val setBySpark = SET_TO_DEFAULT_VALUES
+    // The GCS connector allows appending a custom suffix to the user-agent string.
+    // To prepend Spark's application information, we read the existing suffix,
+    // add our prefix, and set the result back as the new suffix.
+    val sparkGcsPrefix = s"apache_spark/${org.apache.spark.SPARK_VERSION} (GPN:apache_spark)"
```

Comment:
Could you explain where is this string comes from, `apache_spark/` ? I'm curious why this is in this format, not `apache-spark` or other strings. Is there any reason?

Reply:
In addition, why do we need `(GPN:apache_spark)` postfix here?

If there is no reason, I'm wondering if we can simplify this by removing `(GPN:apache_spark)`.

Reply:
Well, not necessarilly. I don't think Apache Spark needs to follow blindly Apache Beam project's convention, @shrutisinghania .